sample-code/src/ai-api/deployment-api.ts

```diff
@@ -0,0 +1,99 @@
+import { DeploymentApi } from '@sap-ai-sdk/ai-api';
+import type {
+  AiDeploymentCreationResponse,
+  AiDeploymentDeletionResponse,
+  AiDeploymentList,
+  AiDeploymentModificationResponse,
+  AiDeploymentResponseWithDetails
+} from '@sap-ai-sdk/ai-api';
+
+/**
+ * Get all deployments.
+ * @param resourceGroup - AI-Resource-Group where the resources are available.
+ * @param status - Optional parameter to filter deployments by status.
+ * @returns All deployments.
+ */
+export async function getDeployments(
+  resourceGroup: string,
+  status?:
+    | 'PENDING'
+    | 'RUNNING'
+    | 'COMPLETED'
+    | 'DEAD'
+    | 'STOPPING'
+    | 'STOPPED'
+    | 'UNKNOWN'
```

Comment:
[pp] The status seems to not be used in the sample. I think a sample is more helpful, when it is concise and specific. For the sample code consider creating an example that filters by a specific status with a specific resource group. Also take a look at the samples for orchestration and azure open ai. Otherwise I am wondering why bother creating this function at all if it just wraps the actual API.

This also applies to the other sample functions here and for the scenario API.

Also note, that in the tests you can use the actual API directly, so you don't need to wrap it here for the tests.